src/rgw/rgw_rest_swift.cc

```diff
@@ -1404,6 +1406,83 @@ RGWOp *RGWHandler_REST_Service_SWIFT::op_delete()
   return NULL;
 }
 
+static bool is_web_mode(const req_state * const s) {
+  const boost::string_ref webmode = s->info.env->get("HTTP_X_WEB_MODE", "");
+  return webmode == "true";
+}
+
+static bool is_website_req(const req_state * const s)
+{
+  /* Static website works only with the GET or HEAD method. Nothing more. */
+  const std::set<boost::string_ref> ws_methods = { "GET", "HEAD" };
+  if (ws_methods.count(s->info.method) == 0) {
+    return false;
+  }
+
+  /* We also need to handle early failures from the authentication system.
+   * In such cases req_state::auth_identity may be empty. Website treats them
+   * the same way as anonymous. */
+  if (! s->auth_identity) {
+    return true;
+  }
+
+  /* Swift serves websites only for anonymous requests unless client explicitly
+   * requested this behaviour by supplying X-Web-Mode HTTP header set to true. */
+  if (s->auth_identity->is_anonymous() || is_web_mode(s)) {
+    return true;
+  }
+
+  return false;
+}
+
+RGWOp* RGWHandler_REST_Bucket_SWIFT::get_ws_index_op()
+{
+  // retarget to get obj
+  s->object = s->bucket_info.website_conf.get_swift_index_doc();
+
+  auto getop = new RGWGetObj_ObjStore_SWIFT;
+  getop->set_get_data(true);
```

Comment:
Swift responses in way typical for `HEAD` - no content inside:
```
$ curl -i -H "X-Auth-Token: $token" "$publicURL/swsc/" -X HEAD -H "X-Web-Mode: True"
HTTP/1.1 200 OK
Content-Length: 1488
Content-Type: text/html; charset=UTF-8
X-Trans-Id: tx44bb8b21a58d4ab2addf9-0057b31de8
Date: Tue, 16 Aug 2016 14:06:32 GMT

```
Thanks for pointing this out! I will differentiate the behavior.